include/lsst/afw/geom/SpanSet.h

```diff
@@ -0,0 +1,868 @@
+     * @param y number of pixels to shift in y dimension
+     */
+    std::shared_ptr<SpanSet> shiftedBy(int x, int y) const;
+    /** @brief Return a new SpanSet shifted by specified amount
+     *
+     * @param offset integer extent which specifies amount to offset in x and y
+     */
+    std::shared_ptr<SpanSet> shiftedBy(Extent2I const & offset) const;
+
+    /** @brief Return a new SpanSet which has all pixel values inside specified box
+     *
+     * @param box Integer box specifying the bounds for which all pixels must be within
+     */
+    std::shared_ptr<SpanSet> clippedTo(Box2I const & box) const;
+
+    /** @brief Return a new SpanSet who's pixels are the product of applying the specified
+               transformation
+     *
+     * @param t A linear transform object which will be used to map the pixels
+     */
+    std::shared_ptr<SpanSet> transformedBy(LinearTransform const & t) const;
+
+    /** @brief Return a new SpanSet who's pixels are the product of applying the specified
+               transformation
+     *
+     * @param t An affine transform object which will be used to map the pixels
+     */
+    std::shared_ptr<SpanSet> transformedBy(AffineTransform const & t) const;
+
+    /** @brief Return a new SpanSet who's pixels are the product of applying the specified
+               transformation
+     *
+     * @param t A XY transform object which will be used to map the pixels
+     */
+    std::shared_ptr<SpanSet> transformedBy(XYTransform const & t) const;
+
+    /** @brief Specifies if this SpanSet overlaps with another SpanSet
+     *
+     * @param other A SpanSet for which overlapping comparison will be made
+     */
+    bool overlaps(SpanSet const & other) const;
+
+    /** @brief Check if a SpanSet instance entirely contains another SpanSet
+     *
+     * @param other The SpanSet who's membership is to be tested for
+     */
+    bool contains(SpanSet const & other) const;
+
+    /** @brief Check if a point is contained within the SpanSet instance
+     *
+     * @param point An integer point object for which membership is to be tested
+     */
+    bool contains(Point2I const & point) const;
+
+    /** @brief Compute the point about which the SpanSet is symmetrically distributed
+     */
+    Point2D computeCentroid() const;
+
+    /** @brief Compute the shape parameters for the distribution of points in the SpanSet
+     */
+    ellipses::Quadrupole computeShape() const;
+
+    /** @brief Perform a set dilation operation, and return a new object
+     *
+     * Dilate a SpanSet with a kernel specified with the stencil parameter
+     *
+     * @param r radius of the stencil, the length is inclusive i.e. 3 ranges from -3 to 3
+     * @param s must be an enumeration of type geom::Stencil. Specifies the shape of the
+                dilation kernel. May be CIRCLE, MANHATTAN, or BOX
+     */
+    std::shared_ptr<SpanSet> dilate(int r, Stencil s = Stencil::CIRCLE) const;
```

Comment:
I'm a little concerned we're calling the `dilate` and `erode` functions the same thing both here and in `Footprint`, where they're in place.  Going by the convention used elsewhere in this file, maybe these should be `eroded` and `dilated`?